src/doc.cpp

```diff
@@ -522,17 +522,19 @@ void Doc::ExportMIDI(smf::MidiFile *midiFile)
         GenerateMIDIFunctor generateScoreDefMIDI(midiFile);
         generateScoreDefMIDI.SetChannel(midiChannel);
         generateScoreDefMIDI.SetTrack(midiTrack);
+
         scoreDef->Process(generateScoreDefMIDI);
 
+        GenerateMIDIFunctor generateMIDI(midiFile);
+        generateMIDI.SetControlEvents(true);
```

Comment:
I'm a bit worried about moving the functor out of the layer loop. This means that when starting the second (or third) layer the functor might still contain some non-default state, e.g. `m_totalTime` is nonzero. I haven't really found an example which is screwed up by this, but I can imagine that this might lead to issues, since the functor state is not properly reset at the end of the layers loop.

This could be avoided by keeping the functor in the loop as before and just creating the controlEvents flag outside the loop.

Reply:
OK, I can change that